test/models/location_test.rb

```diff
@@ -1,6 +1,62 @@
 require 'test_helper'
 require 'models/shared/taxonomies_base_test'
+require 'rfauxfactory'
 
 class LocationTest < ActiveSupport::TestCase
   include TaxonomiesBaseTest
+  valid_loc_data_list = [
+    RFauxFactory.gen_alpha(1),
+    RFauxFactory.gen_alpha(246),
+    *RFauxFactory.gen_strings(1..246, exclude: [:html]).values,
+    RFauxFactory.gen_html(rand((1..221)))
+  ]
+  invalid_loc_name_list = [
+    '',
+    ' ',
+    "\t",
+    *RFauxFactory.gen_strings(247).values
+  ]
+
+  test 'should create with multiple valid names' do
+    valid_loc_data_list.each do |name|
+      location = FactoryBot.create(:location, :name => name)
```

Comment:
I think we do not need to save the model which, SQL queries are slow and we only want to ensure that validation catches it. If you write the test like tihs, it will have the same impact but will be faster since it only builds in-memory objects and performs validation only. Also it could be more fragile if the save would start failing because of other reasons than name validity.

```ruby
location = FactoryBot.build(:location, :name => name)
assert location.valid?
```

more technical details (but I guess you already know since I saw the test with invalid name):
In rails, when you call save it calls valid? inside and does not trigger the SQL query if it returns false. valid? runs all models validations and returns false if some of these registered some error on this model. You could argue that your test is more thorough since it also verifies behavior of save, but we should rely on tests of rails framework that covers it already.